articles/choosing-between-firebase-and-mongodb/index.md

```diff
@@ -0,0 +1,95 @@
+The quantity of data stored and retrieved from the application is one of the factors considered when choosing a database.
+Other factors considered when choosing the right database for an application or a website are:
+1. Latency.
+Latency refers to the response time of the database and to the end-to-end response time of the application. 
+2.structure of the data. Choosing the right database for an application or a website involves the developer understanding the right data structures for storage and retrieval of data. Various applications deal with data of various formats In this article, I am going to focus on Firebase and MongoDB while highlighting its differences.
+### History of MongoDB
+10gen company which is now MongoDB Inc developed MongoDB in the year 2007. The 10gen main language was server-side JavaScript.10gen company started to develop as a Platform as a service ( PAAS) product, but they experienced scalability issues with the existing relational database systems. They developed a document-oriented database system called MongoDB.
+### Introduction to MongoDB
+MongoDB is an open-source document-oriented database that stores data in JSON-like formats. MongoDB makes use of documents and collections instead of using tables and rows. It supports the dynamic schema design. MongoDB is a document-oriented server developed in a C++ programming language.
+### Features of MongoDB
+1. Native Replication.
+MongoDB provides a replication feature by distributing data across different machines through a replica set topology.According to [Congruent Solutions](https://www.congruentsolutions.com/blogposts/mongodb-replication-approach-setup-using-arbiter/),A replica set is a group of MongoDB instances that host the same data set.In a replica set one node is primary node that receives all write operations and all other nodes acts as secondaries. Replica set can have only one primary node.After each successive writes to the primary node, MongoDB will replicate the same data to all the secondary nodes present in the replica set.
+2. Supports Ad-hoc Queries.
+MongoDB supports ad-hoc queries by indexing the Binary JSON ( BSON) documents.
+3. It is schemaless.
+MongoDB is considered flexible since its database is written in the C++ programming language.
+4. High Performance.
+MongoDB's performance is high as compared to other databases due to its scalability, and it has a faster query response because of replication and indexing thus making it a better choice for real-time applications.
+5. Indexing.
+MongoDB indexes improve search performance.
+### Advantages of MongoDB
+1. It contains heterogeneous data.
+2. It provides high performance, scalability, and availability.
+3. It can be integrated with Big Data Hadoop.
+4. MongoDB's documentation has an extensive collection of literature and MongoDB tutorials for new users.
+5. MongoDB has high data security since there is no risk of SQL injection.
```

Comment:
"high data security" = awkward phrasing. Rephrase.

Why is there no risk of SQL injection? Elaborate...